mmctools/helper_functions.py

```diff
@@ -540,31 +543,47 @@ def model4D_spectra(ds,spectra_dim,average_dim,vert_levels,horizontal_locs,fld,f
     fs = 1 / dt
     overlap = 0
     win = hamming(nblock, True) #Assumed non-periodic in the spectra_dim
-    Puuf_cum = np.zeros((len(vert_levels),len(horizontal_locs),ds.dims[spectra_dim]))
+    
+    init_Puuf_cum = True
```

Comment:
I don't follow this, @phawbeck --why not just initialize here?

Reply:
The only difference was the last dimension was set based the original dataset whereas if you initialize it after doing the first power_spectral_density call you get exactly the length you're expecting from the PSD output. I think I ran into an issue when changing this from return_onesided=False to True. This way of initializing just worked more reliably.